jobsdb/migration.go

```diff
@@ -218,33 +270,53 @@ func (jd *HandleT) cleanupStatusTables(ctx context.Context, dsList []dataSetT) e
 
 	return jd.WithTx(func(tx *Tx) error {
 		for _, statusTable := range toCompact {
+			table := statusTable.JobStatusTable
+			// clean up and vacuum if not present in toVacuum
+			_, ok := toVacuum[table]
 			if err := jd.cleanStatusTable(
 				ctx,
 				tx,
-				statusTable.JobStatusTable,
+				table,
+				!ok,
 			); err != nil {
 				return err
 			}
 		}
+		// vacuum full if present in toVacuum
+		for table := range toVacuum {
+			if _, err := tx.ExecContext(ctx, fmt.Sprintf(`VACUUM FULL %[1]q`, table)); err != nil {
+				return err
+			}
+		}
```

Comment:
If more than 1 status tables need full vacuuming, it would be preferable for it to happen in ascending dataset order instead of randomly (map iteration order is random).

That being said, `getVacuumFullCandidates` could return a slice instead of a map & `cleanupStatusTables` could prepare a map from it for its own purposes